tests/composite_pk/test_filter.py

```diff
@@ -476,6 +477,28 @@ def test_outer_ref_pk(self):
                 queryset = Comment.objects.filter(**{f"id{lookup}": subquery})
                 self.assertEqual(queryset.count(), expected_count)
 
+    def test_outer_ref_pk_filter_on_pk(self):
+        subquery = Subquery(User.objects.filter(pk=OuterRef("pk")).values("pk")[:1])
+        tests = [
+            ("exact", 2),
+            ("gt", 0),
+            ("gte", 2),
+            ("lt", 0),
+            ("lte", 2),
+        ]
+        for lookup, expected_count in tests:
+            with self.subTest(f"pk__{lookup}"):
+                qs = Comment.objects.filter(**{f"pk__{lookup}": subquery})
+                if lookup != "exact" and connection.vendor == "oracle":
+                    with self.assertRaisesMessage(
+                        NotSupportedError,
+                        f"{lookup} "
+                        "for composite fields is not supported on this backend",
+                    ):
+                        qs.count()
+                else:
+                    self.assertEqual(qs.count(), expected_count)
```

Comment:
Maybe we should this to two separate tests where one is skipped if `supports_tuple_comparison_subquery` and the other only run if available?